addons/account/wizard/account_automatic_entry_wizard.py

```diff
@@ -170,7 +171,15 @@ def _get_move_dict_vals_change_account(self):
 
             counterpart_balances[(line.partner_id, counterpart_currency)]['amount_currency'] += counterpart_amount_currency
             counterpart_balances[(line.partner_id, counterpart_currency)]['balance'] += line.balance
-            counterpart_balances[(line.partner_id, counterpart_currency)]['analytic_distribution'] = line.analytic_distribution
+            if line.analytic_distribution:
+                for account_id, distribution in line.analytic_distribution.items():
+                    # For the counterpart, we will need to make a prorata of the different distribution of the lines
+                    # This computes the total balance for each analytic account, for each counterpart line to generate
+                    counterpart_distribution_amount[(line.partner_id, counterpart_currency)][account_id] = (
+                        line.balance * distribution
+                        + counterpart_distribution_amount[line.partner_id, counterpart_currency].get(account_id, 0) * 100
+                    ) / 100
+            counterpart_balances[(line.partner_id, counterpart_currency)]['analytic_distribution'] = counterpart_distribution_amount[line.partner_id, counterpart_currency] or {}
```

Comment:
```suggestion
            counterpart_balances[(line.partner_id, counterpart_currency)]['analytic_distribution'] = counterpart_distribution_amount[(line.partner_id, counterpart_currency)] or {}
```

Reply:
(line.partner_id, counterpart_currency) is everywhere... what about storing the key in a separated variable?